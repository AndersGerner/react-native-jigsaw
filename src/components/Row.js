/* @flow */

import * as React from "react";
import { View, Text, Image, StyleSheet } from "react-native";
import { withTheme } from "../core/theming";

type Props = {
  left: Function,
  right: Function,
  style?: any
};

class Row extends React.Component<Props> {
  render() {
    const {
      left,
      right,
      style,
      theme: { spacing }
    } = this.props;

    return (
      <View style={[styles.container, { padding: spacing.large }, style]}>
        {left && left()}
        {right && right()}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  }
});

export default withTheme(Row);
