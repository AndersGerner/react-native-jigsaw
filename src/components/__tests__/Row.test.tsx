import * as React from "react";
import renderer from "react-test-renderer";
import Row from "../Row";

it("renders row", () => {
  const tree = renderer.create(<Row />).toJSON();

  expect(tree).toMatchSnapshot();
});
