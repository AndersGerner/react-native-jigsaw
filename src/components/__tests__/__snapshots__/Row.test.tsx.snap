// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders row 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "padding": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            undefined,
            Object {
              "color": undefined,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
